spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageProtocolResolver.java

```diff
@@ -73,6 +73,17 @@ public void setResourceLoader(ResourceLoader resourceLoader) {
 	@Override
 	public Resource resolve(String location, ResourceLoader resourceLoader) {
 		if (location.startsWith(PROTOCOL)) {
+
+			String pathWithoutProtocol = location.substring(PROTOCOL.length());
+
+			String[] pathParts = pathWithoutProtocol.split("/");
+
+			if (pathParts.length == 0 || pathWithoutProtocol.isEmpty()) {
```

Comment:
Could be nice to throw an exception here saying the path is invalid?
Spring's `Assert` library is nice for that.

Reply:
Actually, I think this function should return null for any situation other than success. Looking at the parent's javadoc, it says something like that, and does not throw something like IOException.